### PR TITLE
Support large file upload

### PR DESCRIPTION
**Pull request by warofworlds**

### CODE REVIEW

File: gateway-api.md [hunk 1]
```diff
@@ -590,22 +605,63 @@ Property | Description
...
+
+Property | Description
+---------|------------
+`gateway.metadata.id` | File ID
+`gateway.metadata.parent.id` | Parent folder ID
+`gateway.metadata.type`| `file`
+`gateway.metadata.name`| File name
+`gateway.metadata.modified` | Millis since the epoch
+`gateway.metadata.file.size`| Total bytes
+`gateway.metadata.file.hash` | Files with the same hash have the same bytes
```
Comment (terrenceyee) [inline gateway-api.md:626]: Update description to match earlier update.

File: gateway-api.md [hunk 2]
```diff
@@ -696,6 +795,119 @@ Status | Description
...
+`gateway.upload.segment.size` | Expected segment size
+
+**RESPONSE**
+
+*JSON*
+
+Property | Description
+---------|------------
+`gateway.upload.id` | Session ID
+`gateway.upload.segment` | List of updated upload segments.
```
Comment (terrenceyee) [inline gateway-api.md:839]: gateway.upload.segments (with an s)?

File: gateway-api.md [hunk 3]
```diff
@@ -696,6 +795,119 @@ Status | Description
...
+
+*JSON*
+
+Property | Description
+---------|------------
+`gateway.upload.id` | Session ID
+`gateway.upload.segment.number` | Sequential segment number starting from 1
+`gateway.upload.segment.sha256` | Segment SHA-256
+`gateway.upload.segment.size` | Segment size
+`gateway.upload.segment.memo` | Opaque gateway information
```
Comment (terrenceyee) [inline gateway-api.md:899]: The response returns different segment information. For procore, an eTag is returned for the segment. It doesn't need to return url and headers. Should the return property also be named "gateway.upload.segment.memo", or something else like "gateway.upload.segment.result"?

File: gateway-api.md [hunk 4]
```diff
@@ -590,22 +605,63 @@ Property | Description
...
 *Body*
 
-The file binary stream.
+Property | Description
+---------|-------------
+`gateway.metadata.name` | New file name
+`gateway.metadata.modified` | New file modified time (Millis since the epoch)
+`gateway.metadata.file.size` | New file size
+`gateway.upload.id` | File upload session ID
+`gateway.upload.segment` | List of updated upload segments
```
Comment (terrenceyee) [inline gateway-api.md:664]: List of updated upload segments from the segmented upload requests.

**Review by terrenceyee — CHANGES_REQUESTED** [review]

File: gateway-api.md (hunk 2 above)
Comment (warofworlds) [reply gateway-api.md:839]: I'm using type as name .... so no s

**Review by terrenceyee — APPROVED** [review]

**Pull request merged**